yang/ietf-acl-enh.yang

```diff
@@ -1,529 +1,634 @@
+        position 0;
+        description
+          "If set, logical negation of operation.";
+      }
+      bit match {
+        position 1;
+        description
+          "Match bit.  This is a bitwise match operation
+           defined as '(data & value) == value'.";
+      }
+      bit any {
+        position 2;
+        description
+          "Any bit.  This is a match on any of the bits in
+           bitmask.  It evaluates to 'true' if any of the bits
+           in the value mask are set in the data,
+           i.e., '(data & value) != 0'.";
+      }
+    }
+    description
+      "Specifies how to apply the defined bitmask.
+       'any' and 'match' bits must not be set simultaneously.";
+  }
+
+  typedef fragment-type {
+    type bits {
+      bit df {
+        position 0;
+        description
+          "Don't fragment bit for IPv4.
+           Must be set to 0 when it appears in an IPv6 filter.";
+      }
+      bit isf {
+        position 1;
+        description
+          "Is a fragment.";
+      }
+      bit ff {
+        position 2;
+        description
+          "First fragment.";
+      }
+      bit lf {
+        position 3;
+        description
+          "Last fragment.";
+      }
+    }
+    description
+      "Different fragment types to match against.";
+  }
+
+  typedef ipv4-prefix-set-ref {
+    type leafref {
+      path "/acl-enh:defined-sets/acl-enh:ipv4-prefix-sets"
+         + "/acl-enh:prefix-set/acl-enh:name";
+    }
+    description
+      "Defines a reference to an IPv4 prefix set.";
+  }
+
+  typedef ipv6-prefix-set-ref {
+    type leafref {
+      path "/acl-enh:defined-sets/acl-enh:ipv6-prefix-sets"
+         + "/acl-enh:prefix-set/acl-enh:name";
+    }
+    description
+      "Defines a reference to an IPv6 prefix set.";
+  }
+
+  typedef port-set-ref {
+    type leafref {
+      path "/acl-enh:defined-sets/acl-enh:port-sets"
+         + "/acl-enh:port-set/acl-enh:name";
+    }
+    description
+      "Defines a reference to a port set.";
+  }
+
+  typedef protocol-set-ref {
+    type leafref {
+      path "/acl-enh:defined-sets/acl-enh:protocol-sets"
+         + "/acl-enh:protocol-set/acl-enh:name";
+    }
+    description
+      "Defines a reference to a protocol set.";
+  }
+
+  typedef icmp-type-set-ref {
+    type leafref {
+      path "/acl-enh:defined-sets/acl-enh:icmp-type-sets"
+         + "/acl-enh:icmp-type-set/acl-enh:name";
+    }
+    description
+      "Defines a reference to an ICMP type set.";
+  }
+
+  grouping tcp-flags {
+    description
+      "Operations on TCP flags.";
+    choice mode  {
+      case explicit {
+        leaf operator {
+          type operator;
+          default "match";
+          description
+            "How to interpret the TCP flags.";
+        }
+        leaf-list explicit-tcp-flag {
+          type identityref {
+            base tcp-flag;
+          }
+          description
+            "An explicit list of the TCP flags that are to be matched. .";
```

Comment:
We need to indicate that this one is inferior to the full bitmask as checks only on known flags may let pass some packets while they carry "weird" flags (e.g.. SYN set but bit position 6 is also set.)